inc/class-statify-frontend.php

```diff
@@ -22,42 +22,42 @@ class Statify_Frontend extends Statify {
 	 * Track the page view
 	 *
 	 * @since    0.1.0
-	 * @version  1.4.2
+	 * @since    1.7.0 $is_snippet parameter added.
+	 * @version  1.7.0
 	 *
-	 * @return   bool
+	 * @param boolean $is_snippet Is tracking triggered via JS (default: false).
+	 *
+	 * @return   boolean
 	 */
-	public static function track_visit() {
+	public static function track_visit( $is_snippet = false ) {
 
-		/* Init vars */
+		// Check of JS snippet is configured.
 		$use_snippet = self::$_options['snippet'];
-		$is_snippet  = $use_snippet && get_query_var( 'statify_target' );
 
-		/* Set target & referrer */
-		if ( $is_snippet ) {
-			$target   = urldecode( get_query_var( 'statify_target' ) );
-			$referrer = urldecode( get_query_var( 'statify_referrer' ) );
+		// Set target & referrer.
+		if ( $use_snippet && $is_snippet ) {
+			$target   = urldecode( isset( $_REQUEST['statify_target'] ) ? wp_unslash( $_REQUEST['statify_target'] ) : '/' );
+			$referrer = urldecode( isset( $_REQUEST['statify_referrer'] ) ? wp_unslash( $_REQUEST['statify_referrer'] ) : '' );
 		} elseif ( ! $use_snippet ) {
-			$target   = filter_var(
-				( isset( $_SERVER['REQUEST_URI'] ) ? wp_unslash( $_SERVER['REQUEST_URI'] ) : '/' ),
-				FILTER_SANITIZE_URL
-			);
-			if ( is_null( $target ) || false === $target ) {
-				$target = '/';
-			} else {
-				$target = wp_unslash( $target );
-			}
-
-			$referrer = filter_var(
-				( isset( $_SERVER['HTTP_REFERER'] ) ? wp_unslash( $_SERVER['HTTP_REFERER'] ) : '' ),
-				FILTER_SANITIZE_URL
-			);
-			if ( is_null( $referrer ) || false === $referrer ) {
-				$referrer = '';
-			}
+			$target   = isset( $_SERVER['REQUEST_URI'] ) ? wp_unslash( $_SERVER['REQUEST_URI'] ) : '/';
+			$referrer = isset( $_SERVER['HTTP_REFERER'] ) ? wp_unslash( $_SERVER['HTTP_REFERER'] ) : '';
 		} else {
 			return false;
 		}
 
+		// Sanitize.
+		$target = filter_var( $target, FILTER_SANITIZE_URL );
+		if ( is_null( $target ) || false === $target ) {
+			$target = '/';
+		} else {
```

Comment:
The `else` is not really necessary, so you should delete that and use only the var if not.

Reply:
Good point, not it's gone.